torch/_inductor/triton_heuristics.py

```diff
@@ -405,7 +407,12 @@ def launcher({', '.join(def_args)}, grid, stream):
 
     def bench(self, launcher, *args, grid, **kwargs):
         """Measure the performance of a given launcher"""
-        if launcher.n_spills > config.triton.spill_threshold:
+        # we don't skip configs wiht spilled registers when auto-tuning custom
+        # (user-written) Triton kernels, as (i) we don't have any knowledge or
+        # control over the kernel code; (ii) there is empirical evidence that
+        # for some (complicated) custom Triton kernels, a register-spilling
+        # config may yield the best latency.
+        if not self.custom_kernel and launcher.n_spills > config.triton.spill_threshold:
```

Comment:
Should we add a config option for this instead of never skipping for custom kernels?

Reply:
I feel, it's fine to never skip custom kernels. Because it's not going to change the result, just delay the auto-tuning. My rationale here is: if the user wants to auto-tune on all configs, we should do that. Triton doesn't skip auto-tuning configs based on inferred number of spilled registers. And to provide similar UX, we should do the same.